pyrate/core/timeseries.py

```diff
@@ -302,6 +303,108 @@ def _solve_ts_lap(nvelpar, velflag, ifgv, mat_b, smorder, smfactor, sel, vcmt):
     return tsvel
 
 
+def linear_rate_pixel(y, t):
+    """
+    Calculate best fitting linear rate to cumulative displacement
+    time series for one pixel using linear regression.
+
+    :param ndarray y: 1-dimensional vector of cumulative displacement time series
+    :param ndarray t: 1-dimensional vector of cumulative time at each epoch
+
+    :return: linrate: Linear rate; gradient of fitted line.
+    :rtype: float
+    :return intercept: Y-axis intercept of fitted line at t = 0
+    :rtype: float
+    :return: rsquared: R-squared value of the linear regression
+    :rtype: float
+    :return: error: Standard error of the linear regression
+    :rtype: float
+    :return: samples: Number of observations used in linear regression
+    :rtype: int
+    """
+
+    # Mask to exclude nan elements
+    mask = ~isnan(y)
+    # remove nan elements from both arrays
+    y = y[mask]    
+    try:
+        t = t[mask]
+    except IndexError:
+        raise TimeSeriesError("linear_rate_pixel: y and t are not equal length")
+
```

Comment:
One more time, sorry! Given `len(y) == len(t)` prior to entering this function as checked in the calling function, is the `IndexError` a possibility?

Reply:
yes, this is there in case someone calls this function directly, and independently of the PyRate workflow. In that situation the could inadvertently pass unequal length vectors.

Reply:
There are no consumers for this though? It should probably be a protected function. Functions that do the heavy lifting/i.e., called billions of time should avoid such interruptions.

Reply:
I see your point but don't think it should be enforced for this PR because I know that this idea is not already consistently applied across PyRate. However, I do think as a result of this discussion that we should set aside some time for a package-wide review of function exposure and protection and do something consistently across PyRate. I'll lodge a ticket in the backlog ;-)